cps/main.c

```diff
@@ -832,23 +832,25 @@ match_bgp4(struct rte_mbuf *pkt, struct gatekeeper_if *iface)
 		rte_pktmbuf_mtod(pkt, struct ether_hdr *);
 	struct ipv4_hdr *ip4hdr;
 	struct tcp_hdr *tcp_hdr;
-	uint16_t minimum_size = sizeof(*eth_hdr) +
+	uint16_t ether_type_be = pkt_in_skip_l2(pkt, eth_hdr, (void **)&ip4hdr);
+	size_t l2_len = pkt_in_l2_hdr_len(pkt);
+	uint16_t minimum_size = l2_len +
 		sizeof(struct ipv4_hdr) + sizeof(struct tcp_hdr);
 	uint16_t cps_bgp_port = rte_cpu_to_be_16(get_cps_conf()->tcp_port_bgp);
 
-	if (unlikely(eth_hdr->ether_type != BE_ETHER_TYPE_IPv4))
+	if (unlikely(ether_type_be != BE_ETHER_TYPE_IPv4))
 		return -ENOENT;
 
 	if (pkt->data_len < minimum_size)
 		return -ENOENT;
 
-	ip4hdr = (struct ipv4_hdr *)&eth_hdr[1];
 	if (ip4hdr->dst_addr != rte_cpu_to_be_32(iface->ip4_addr.s_addr))
```

Comment:
Isn't `iface->ip4_addr.s_addr` already in network order?

Reply:
Any update on this question?

Reply:
Yes, it should be kept in network order.